src/ResourceManager/Compute/Commands.Compute/Strategies/ComputeRp/SubResourceStrategy.cs

```diff
@@ -0,0 +1,36 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Strategies;
+using Microsoft.Azure.Management.Compute.Models;
+using N = Microsoft.Azure.Management.Internal.Network.Version2017_10_01.Models;
+
+namespace Microsoft.Azure.Commands.Compute.Strategies.ComputeRp
+{
+    static class SubResourceStrategy
+    {
+        static SubResource GetSuResourceReference(
+            this IEngine engine, IEntityConfig config)
+            => config == null ? null : new SubResource { Id = engine.GetId(config) };
+
+        public static SubResource GetReference(
+            this IEngine engine, ResourceConfig<AvailabilitySet> availabilitySet)
+            => engine.GetSuResourceReference(availabilitySet);
```

Comment:
should this method name be GetSubResourceReference?

Reply:
The idea is to use always the same function name for all resource to get a reference. Different overloads will return different results. For example:
- Network: 
  - uses the same types for a resource Id wrapper object. So the function is `TModel GetReference(ResourceConfig<TModel>)`
  - uses `SubResource` for a nested resource Id wrapper object. `TModel GetReference(NestedResourceConfig<TModel>) where TModel : SubResource`
- Compute uses all kind of Id wrapper objects for different resource types, so I provided several `GetReference` overloads.